spring-integration-core/src/main/java/org/springframework/integration/dsl/RecipientListRouterSpec.java

```diff
@@ -89,8 +97,14 @@ public RecipientListRouterSpec recipientMessageSelector(String channelName, Mess
 	 * @return the router spec.
 	 */
 	public <P> RecipientListRouterSpec recipient(String channelName, GenericSelector<P> selector) {
-		Assert.hasText(channelName);
-		((DslRecipientListRouter) this.target).add(channelName, selector);
+		MessageSelector messageSelector;
+		if (selector instanceof MessageSelector) {
+			messageSelector = (MessageSelector) selector;
+		}
+		else {
+			messageSelector = new MethodInvokingSelector(new LambdaMessageProcessor(selector, null));
```

Comment:
Is this strictly correct? Don't we have to check it's a Lambda first? Someone might implement his own `GenericSelector<P>` that is not a `MessageSelector`.